### PR TITLE
Implementation of sort/argsort and set functions per array API spec

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
Implemented data-parallel merge-sort algorithm, which powers array API operations `dpct.tensor.sort` and `dpctl.tensor.argsort`. Implemented set functions: `dpctl.tensor.unique_values`, `dpctl.tensor.unique_counts`, `dpctl.tensor.unique_inverse`, and `doctl.tensor.unique_all`.

- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to an issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [x] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you opening the PR as a draft?

### CODE REVIEW

**Comment by ndgrigorian** [comment]
@oleksandr-pavlyk 
 Tests seem to pass using the asynchronous `unique_*` functions. Should I go ahead and remove `set_functions_sync.py` and transition to the asynchronous implementations?

**Comment by oleksandr-pavlyk** [comment]
@ndgrigorian I have switched to asynchronous implementation per your feedback, but kept the synchronous one around just in case.

File: dpctl/tensor/_set_functions_async.py [hunk 1]
```diff
@@ -94,7 +96,7 @@ def unique_values(x: dpt.usm_ndarray) -> dpt.usm_ndarray:
         )
         host_tasks.append(ht_ev)
         ht_ev, sort_ev = _sort_ascending(
-            src=fx,
+            src=tmp,
```
Comment (oleksandr-pavlyk) [inline dpctl/tensor/_set_functions_async.py:99]: 👍 Good catch!

**Review by ndgrigorian — APPROVED** [review]
LGTM, thank you @oleksandr-pavlyk , very close to full array API coverage.

**Review by vtavana — APPROVED** [review]
Thank you, works fine for zero dimensional arrays and arrays with `nan`.

**Pull request merged**